src/lib/net/classes/remote.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * !TODO: Add server/client differentation, yeah I know this should have made from the very first start, but lol,
+ * that will take a bit of my brain, but i already got something planned out for this.
+ */
+
+import { BaseEvent } from "../types";
+
+export declare enum EAction {
+	COND = "condition",
+	DO = "do",
+}
+
+function isAFunction(func: (() => boolean) | boolean): func is Callback {
+	return typeOf(func) === "function";
+}
+
+export class RemoteClass<T extends unknown[]> implements BaseEvent<T> {
```

Comment:
Let's refine the vocabulary to `Endpoint` instead of `Remote`

Reply:
Sounds good to me!